### ISSUE TITLE
Typescript 4.7 - nodenext wrong errors on imports containing searchParam

### ISSUE THREAD

**Issue by frank-dspeed** [microsoft/TypeScript#48725]
# Bug Report

### 🔎 Search Terms

searchParam nodenext import needs extensions

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about searchParms in imports

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5JA)

### 💻 Code

```ts
import { Blob } from "./cblite.mjs?cblite=../out/native/";
// Relative Importpath needs extension in EcmaScript-Imports, when „--moduleResolution“ auf „node12“ oder „nodenext“ add extension .ts(2834)
```

### 🙁 Actual behavior

Errors i should add a extension 

### 🙂 Expected behavior

i added a extension and the parameter should be of type url so all is valid ECMAScript.

this comes because of this implementation which is also Valid ECMAScript

cblite.mjs
```js
const importMeta = import.meta.url;
const pathToNative = new URL(importMeta).searchParams.get("cblite");
export const Blob = {};
```
